File: 713-subarray-product-less-than-k/subarray-product-less-than-k.java
class Solution {

    public int numSubarrayProductLessThanK(int[] nums, int k) {
        int ans = 0;
        int mul = 1;
        for(int r=0, f=0; r<nums.length; r++){
            mul *= nums[r];
            while (mul >= k && f <= r) mul /= nums[f++];
            ans += r-f+1;
        }
        return ans;
    }
}

File: 713-subarray-product-less-than-k/subarray-product-less-than-k.cpp
class Solution {
public:

int numSubarrayProductLessThanK(vector<int>& nums, int k) {
    int ans(0), mul(1);
    for(int r=0, f=0; r<nums.size(); r++){
        mul *= nums[r];
        while(mul >= k && f <= r) mul /= nums[f++];
        ans += max(0,(r-f+1));
            // cout <<"r " << r << " f " << f << " len " << (r-f+1) << "\n";
    }
    return ans;
}
};